### PR TITLE
[lxd]: fix container_create not creating devices other than of type disk

### LINKED ISSUE #63996: [BUG] lxd.container_create doesn't create devices correctly
**Description**
Creating a container with `lxd.container_create` with the `devices=` parameter will fail if a device is not of type `disk`.

**Steps to Reproduce the behavior**
1. Create a container with a proxy device:
```
salt-call lxd.container_create name="test" source="jammy/amd64" devices="{test_device: {type: proxy, connect: "tcp:127.0.0.1:9999", listen: "tcp:127.0.0.1:9998"}}"
```
2. Inspect the created container:
```
lxc config show test
```
3. Device wasn't created, instead error message is shown:
```
Passed invalid arguments: path must be given as parameter.
```

**Expected behavior**
Devices are correctly added to the container.

**Versions Report**
any from at least 3001

### PR DESCRIPTION
**Pull request by goapunk**
### What does this PR do?
Extract the `type` argument into `device_type` so it's not overwritten by `container_device_add`.

### What issues does this PR fix or reference?
Fixes: #63996 

### Previous Behavior
`lxd.container_create` wouldn't create devices with type other than `disk`.

### New Behavior
`lxd.container_create` correctly creates devices with any type.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by goapunk** [comment]
@jochumdev tagging you as you are still listed as maintainer (?).

File: salt/modules/lxd.py
Comment (Ch3LL) [inline salt/modules/lxd.py:693]:
```suggestion
                container_device_add(name, dn, device_type=dargs["type"], **dargs)
```

Just want to ensure kwargs are kept in tact in case someone changes the order.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
This will also require test coverage.

File: tests/integration/states/test_lxd_container.py
Comment (goapunk) [inline tests/integration/states/test_lxd_container.py:36]: ensure containers are torn down if still running

**Review by Ch3LL — APPROVED** [review]

**Comment by goapunk** [comment]
@Ch3LL is there anything blocking this from being merged? Do I need to do anything?

**Pull request merged**
